---
title: "E16_Dec_subset"
output: html_notebook
---

```{r}
#load("E16_Dec7v3_Trachea.RData")
```

##### basal, secretory, and ciliated:
```{r}
E16_Dec7v3_Trachea <- SetAllIdent(object = E16_Dec7v3_Trachea, id = "res.1.2")
E16_Dec7v3_epi<-SubsetData(object=E16_Dec7v3_Trachea,ident.use=c(1,20,17,13,19,6,9,4,7))
table(E16_Dec7v3_epi@meta.data$res.1.2)
```
```{r}
colnames(E16_Dec7v3_epi@meta.data)[colnames(E16_Dec7v3_epi@meta.data) == 'res.0.8'] <- 'orig.0.8'
colnames(E16_Dec7v3_epi@meta.data)[colnames(E16_Dec7v3_epi@meta.data) == 'res.1.2'] <- 'orig.1.2'
```
```{r}
E16_Dec7v3_epi <- ScaleData(object = E16_Dec7v3_epi)
```
```{r}
E16_Dec7v3_epi <- FindVariableGenes(object = E16_Dec7v3_epi, do.plot = TRUE, x.low.cutoff=0.1,x.high.cutoff = Inf, y.cutoff = 0.5)
```
######run PCA on the set of genes
```{r}
E16_Dec7v3_epi <- RunPCA(object = E16_Dec7v3_epi, do.print = FALSE)
#PCAPlot(E16_Dec7v3_epi)
```

```{r}
E16_Dec7v3_epi <- ProjectPCA(object = E16_Dec7v3_epi, do.print = F)
```

```{r}
PCElbowPlot(object = E16_Dec7v3_epi)
```
```{r,fig.height=50,fig.width=15}
PCHeatmap(object = E16_Dec7v3_epi, pc.use = 1:20, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, num.genes = 25)

```


```{r}
n.pcs.sub = 17
```

```{r}
res.used <- 1.2
```

```{r}
E16_Dec7v3_epi <- FindClusters(object = E16_Dec7v3_epi, reduction.type = "pca", dims.use = 1:n.pcs.sub, 
                     resolution = res.used, print.output = 0, force.recalc = T)
```
```{r}
E16_Dec7v3_epi <- RunTSNE(object = E16_Dec7v3_epi, dims.use = 1:n.pcs.sub, perplexity=30)
```
```{r, fig.width=10,fig.height=9}
TSNEPlot(object = E16_Dec7v3_epi, do.label = T,pt.size = 0.4)
```

##### get tSNE embedding for velocyto:
```{r}
E16_Dec_cv3_ID<-read.csv(file="E16_Dec_cv3_ID.csv",header=F,sep=",",stringsAsFactors = F) 

```
```{r}
head(E16_Dec_cv3_ID)

```

```{r}
E16_Dec_cv3_name<-gsub("x","",E16_Dec_cv3_ID)
head(E16_Dec_cv3_name)
```
```{r}
E16_Dec_cv3_name<-gsub(":","_",E16_Dec_cv3_name)
head(E16_Dec_cv3_name)
```
```{r}
TSNE1_Loomorder_epi<-E16_Dec7v3_epi@dr$tsne@cell.embeddings[match(E16_Dec_cv3_name,rownames(E16_Dec7v3_epi@dr$tsne@cell.embeddings)),1]
write(TSNE1_Loomorder_epi,"TSNE1_Loomorder_epi.csv",ncolumns=1,sep=",")
```
```{r}
head(TSNE1_Loomorder_epi)
```
```{r}
TSNE2_Loomorder_epi<-E16_Dec7v3_epi@dr$tsne@cell.embeddings[match(E16_Dec_cv3_name,rownames(E16_Dec7v3_epi@dr$tsne@cell.embeddings)),2]
write(TSNE2_Loomorder_epi,"TSNE2_Loomorder_epi.csv",ncolumns=1,sep=",")

```
##### markers for each ciliated population:
```{r}
E16_Dec_epi_res12_8over1011<-FindMarkers(E16_Dec7v3_epi,ident.1=c(8),ident.2 = c(10,11),only.pos = TRUE)
E16_Dec_epi_res12_8over1011
```
```{r}
write.table(E16_Dec_epi_res12_8over1011,"epiSubset_c8inCilia.txt",sep="\t")

```


```{r}
E16_Dec_epi_res12_10over811<-FindMarkers(E16_Dec7v3_epi,ident.1=c(10),ident.2 = c(8,11),only.pos = TRUE)
E16_Dec_epi_res12_10over811
```
```{r}
E16_Dec_epi_res12_10over8<-FindMarkers(E16_Dec7v3_epi,ident.1=c(10),ident.2 = c(8),only.pos = TRUE)
E16_Dec_epi_res12_10over8
```
```{r}
write.table(E16_Dec_epi_res12_10over8,"epiSubset_c10overC8.txt",sep="\t")

```

```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.2")

E16_Dec_epi_res12_10over11<-FindMarkers(E16_Dec7v3_epi,ident.1=c(10),ident.2 = c(11),only.pos = TRUE)
E16_Dec_epi_res12_10over11
```
```{r}
write.table(E16_Dec_epi_res12_10over11,"epiSubset_c10overC11.txt",sep="\t")

```

```{r}
E16_Dec_epi_res12_11over810<-FindMarkers(E16_Dec7v3_epi,ident.1=c(11),ident.2 = c(8,10),only.pos = TRUE)
E16_Dec_epi_res12_11over810
```

```{r}
write.table(E16_Dec_epi_res12_11over810,"epiSubset_c11inCilia.txt",sep="\t")

```



```{r,fig.height=4,fig.width=16}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.2")

DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Foxj1","Top2a","Mcidas","Ccno","Foxn4","Shisa8","Lrrc23","Prr18","Cfap53","Cdhr3","Sntn","Ifitm1","Lbp","Ly6c1","Ly6a"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = F,cells.use = E16_Dec7v3_epi@cell.names[E16_Dec7v3_epi@meta.data$res.1.2 %in% c(8,10,11)],group.order = c(8,10,11),group.cex = 30,cex.row = 20
  )
```


##### scoring:
```{r,fig.height=4,fig.width=28}
E16_Dec7v3_epi@data[1:6,1:6]

```

```{r,fig.height=4,fig.width=28}
percentile_table_epi<-apply(E16_Dec7v3_epi@data,1,percent_rank)

```
```{r,fig.height=4,fig.width=28}
percentile_table_epi[1:6,1:6]

```


```{r}
OMIMgene<-read.csv(file = "genesOMIM.csv",header=T,sep=",",stringsAsFactors = F)
OMIMgene<-lapply(OMIMgene,function(x) unlist(strsplit(unlist(x),split=","))) 
head(OMIMgene$Mucociliary)
```
```{r}
OMIMgene_mucosaGoblet<-as.vector(read.csv(file = "genesOMIM_mucosa_goblet.csv",header=T,sep=",",stringsAsFactors = F)[,1])
OMIMgene_mucosaGoblet<-unlist(strsplit(unlist(OMIMgene_mucosaGoblet),split=","))
OMIMgene_mucosaGoblet[90:105]
```
```{r}
 mocosaGoblet_score<- apply(percentile_table_epi[,colnames(percentile_table_epi) %in% OMIMgene_mucosaGoblet],1,mean)
```
```{r}
 head( mocosaGoblet_score)
```
```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = mocosaGoblet_score, col.name = "mocosaGoblet_score")

```
```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("mocosaGoblet_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.4")

```





```{r}
 ciliopathy_table<- percentile_table_epi[,colnames(percentile_table_epi) %in% OMIMgene$Ciliopathy]
```

```{r}
 ciliopathy_score<- apply(ciliopathy_table,1,mean)
```
```{r}
 head(ciliopathy_score)
```
```{r}
 PCD_score<- apply(percentile_table_epi[,colnames(percentile_table_epi) %in% OMIMgene$Primary.ciliary.dyskinesia],1,mean)
```
```{r}
 head(PCD_score)
```

```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = ciliopathy_score, col.name = "ciliopathy_score")

```
```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = PCD_score, col.name = "PCD_score")

```
```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("ciliopathy_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.2")

```
```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("PCD_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.2")

```
```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("PCD_score"), nCol = 1,ident.include = c(8,10,11),x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.2")

```

```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.2")

VlnPlot(object = E16_Dec7v3_epi, features.plot = c("ciliopathy_score"), ident.include = c(11),nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="seq_group")

```
```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.2")

VlnPlot(object = E16_Dec7v3_epi, features.plot = c("PCD_score"), ident.include = c(8),nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="seq_group")

```
```{r}
 mucus_score<- apply(percentile_table_epi[,colnames(percentile_table_epi) %in% c(OMIMgene$Airway...Mucus,OMIMgene$Pulmonary.and.Mucus)],1,mean)
```
```{r}
 head(mucus_score)
```
```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = mucus_score, col.name = "mucus_score")

```

```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("mucus_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.4")

```
```{r}
 COPD_score<- apply(percentile_table_epi[,colnames(percentile_table_epi) %in% c(OMIMgene$COPD)],1,mean)
```
```{r}
 head(COPD_score)
```
```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = COPD_score, col.name = "COPD_score")

```

```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("COPD_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.4")

```

```{r}
 asthma_score<- apply(percentile_table_epi[,colnames(percentile_table_epi) %in% c(OMIMgene$Pulmonary...Asthma)],1,mean)
```
```{r}
 head(asthma_score)
```
```{r}
E16_Dec7v3_epi<-AddMetaData(object = E16_Dec7v3_epi, metadata = asthma_score, col.name = "asthma_score")

```

```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("asthma_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.4")

```
```{r, fig.height=5, fig.width=14}
ggplot(E16_Dec7v3_epi@meta.data,aes(genotype,asthma_score))+facet_grid(.~res.1.2)+geom_dotplot(binaxis="y",aes(color=genotype,fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1))
```




```{r}
res.used <- 1.4
```
```{r}
E16_Dec7v3_epi <- FindClusters(object = E16_Dec7v3_epi, reduction.type = "pca", dims.use = 1:n.pcs.sub, 
                     resolution = res.used, print.output = 0, force.recalc = T)
```
```{r}
E16_Dec7v3_epi <- RunTSNE(object = E16_Dec7v3_epi, dims.use = 1:n.pcs.sub, perplexity=30)
```
```{r, fig.width=10,fig.height=9}
TSNEPlot(object = E16_Dec7v3_epi, do.label = T,pt.size = 0.4)
```


```{r,fig.height=8,fig.width=22}
E16_Dec7v3_epi=buildClusterTree(E16_Dec7v3_epi,do.reorder = F,reorder.numeric = F,pcs.use = 1:17)

```


```{r}
table(E16_Dec7v3_epi@meta.data$res.1.4,E16_Dec7v3_epi@meta.data$seq_group)
```

```{r}
prop.table(table(E16_Dec7v3_epi@meta.data$res.1.4,E16_Dec7v3_epi@meta.data$seq_group),2)
```


```{r,fig.height=9,fig.width=28}

DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Epcam","Trp63","Krt5","Sostdc1","Col6a1","Col6a2","Col6a3","Bgn","Postn","Tk1","Mki67","Top2a","Creb3l1","Muc5ac","Gp2","Rac2","1810010H24Rik","Krt15","Krt4","Krt13","mt-Co3","mt-Cytb","Galnt6","Ptgdr","B3gnt6","Cd177","Foxn4","Foxj1","Cdhr3","Ano1"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,group.by="res.1.4",cex.row = 20
  )
```
```{r}
VlnPlot(object = E16_Dec7v3_epi, features.plot = c("doublet_score"), nCol = 1,x.lab.rot = T,point.size.use = 0.3,use.raw=F,group.by="res.1.4")

```
```{r}
E16_Dec7v3_epi@meta.data$cell_type<-mapvalues(E16_Dec7v3_epi@meta.data$res.1.4,from=c("0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"),to=c("Basal","Basal","Secretory","Basal/Secretory","Secretory","Secretory","Secretory","Ciliated","Secretory","Secretory","Secretory","Ciliated","Secretory","Basal","Doublet","Ciliated","Secretory","Doublet","Doublet"))
```


##### c14, 17 and c18 are doublets.
```{r,fig.width=5,fig.height=5}
ggplot(data=E16_Dec7v3_epi@meta.data[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18)),],aes(seq_group,fill=cell_type))+ 
    geom_bar(position="fill")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r,fig.width=5,fig.height=5}
ggplot(data=E16_Dec7v3_epi@meta.data[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18)),],aes(genotype,fill=cell_type))+ 
    geom_bar(position="fill")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
```{r,fig.width=5,fig.height=5}
table(E16_Dec7v3_epi@meta.data$cell_type[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))],E16_Dec7v3_epi@meta.data$genotype[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))])
```
```{r}
DE_E16_secretory_genotype<-FindMarkers(E16_Dec7v3_epi,cells.1<-WhichCells(object=E16_Dec7v3_epi,cells.use = (E16_Dec7v3_epi@meta.data$genotype=="wt" & E16_Dec7v3_epi@meta.data$cell_type=="Secretory" )),cells.2<-WhichCells(object=E16_Dec7v3_epi,cells.use = (E16_Dec7v3_epi@meta.data$genotype=="mut" & E16_Dec7v3_epi@meta.data$cell_type=="Secretory" )),only.pos = F,logfc.threshold=0,min.pct=0)
DE_E16_secretory_genotype
```
```{r}
library(ggrepel)
```
```{r}
DE_E16_secretory_genotype$gene<-rownames(DE_E16_secretory_genotype)
DE_E16_secretory_genotype$sig<-DE_E16_secretory_genotype$p_val_adj<0.001
volc = ggplot(DE_E16_secretory_genotype, aes(avg_logFC, -log10(p_val_adj))) + #volcanoplot with avg_logFC versus p_val_adj
    geom_point(aes(col=sig)) + #add points colored by significance
    scale_color_manual(values=c("black", "red")) + 
    ggtitle("E16secretory_wt/mut") + geom_text_repel(data=head(DE_E16_secretory_genotype, 40), aes(label=gene), point.padding = 1, box.padding = .3) +
  labs(y = expression(-log[10]*" "*"adjusted pvalue"), x = "avg log fold change") + 
  theme(legend.title = element_blank(), legend.position = "top") + 
  scale_fill_discrete(labels = c("Not Sig", "adjusted pval < 0.001"))
```
```{r}
volc
```
```{r}
E16_Dec7v3_epi@meta.data$specific_type<-mapvalues(E16_Dec7v3_epi@meta.data$res.1.4,from=c("0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"),to=c("Basal-Sostdc1","Basal","Secretory-Krt4","Basal/Secretory","Secretory-Krt4","Secretory","Secretory-Krt4","Ciliated","Secretory","Secretory-Krt4","Secretory-Krt4","Ciliated","CyclingSecretory","CyclingBasal","Doublet","Ciliated","Secretory","Doublet","Doublet"))
```


```{r,fig.width=5,fig.height=5}
table(E16_Dec7v3_epi@meta.data$specific_type[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))],E16_Dec7v3_epi@meta.data$genotype[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))])
```

```{r,fig.width=5,fig.height=5}
ggplot(data=E16_Dec7v3_epi@meta.data[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18)),],aes(genotype,fill=specific_type))+ 
    geom_bar(position="fill")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
##### markers for specific clusters:
```{r,fig.height=3,fig.width=8}
E16_Dec7v3_epi<-SetAllIdent(object = E16_Dec7v3_epi, id = "specific_type")

E16_Dec7v3_epi@ident=factor(E16_Dec7v3_epi@ident,levels(E16_Dec7v3_epi@ident)[c(4,8,9,2,1,3,6,5,7)])
DotPlot(object = E16_Dec7v3_epi, cols.use = c("lightgrey","red"),genes.plot = c("Foxj1","Ptgdr","B3gnt6","Galnt6","Cgref1","Gp2","Tff2","Muc5b","Muc16","Cited1","Krt4","Creb3l1","Spdef","Clic3","Ccl20","Sostdc1","Smoc2","Krt14","Bmp7","Trp63","Krt5","Mki67","Top2a"),group.by = "ident", x.lab.rot = T,plot.legend = T)
```

```{r,fig.height=3,fig.width=8}
E16_Dec7v3_epi<-SetAllIdent(object = E16_Dec7v3_epi, id = "specific_type")

E16_Dec7v3_epi@ident=factor(E16_Dec7v3_epi@ident,levels(E16_Dec7v3_epi@ident)[c(4,8,9,2,1,3,6,5,7)])
DotPlot(object = E16_Dec7v3_epi, cols.use = c("forestgreen","magenta3"),genes.plot = c("Foxj1","Ptgdr","B3gnt6","Galnt6","Cgref1","Gp2","Tff2","Muc5b","Muc16","Cited1","Krt4","Creb3l1","Spdef","Clic3","Ccl20","Sostdc1","Smoc2","Krt14","Bmp7","Trp63","Krt5","Mki67","Top2a"),group.by = "ident", x.lab.rot = T,plot.legend = T,col.max = 2,col.min = -2)
```


```{r,fig.height=3,fig.width=8}
print(levels(E16_Dec7v3_epi@ident))
```


```{r,fig.height=6,fig.width=28}

DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Trp63","Krt5","Krt14","Sostdc1","Tk1","Mki67","Top2a","Spdef","Creb3l1","Muc5b","Muc5ac","Gp2","Krt15","Krt4","Krt13","Foxn4","Mcidas","Foxj1","Cdhr3","Ano1"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,group.by="res.1.4",cells.use = E16_Dec7v3_epi@cell.names[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))]
  )
```
```{r,fig.height=15,fig.width=45}

DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Trp63","Krt5","Krt14","Sostdc1","Tk1","Mki67","Top2a","Sftpb","Krt4","Krt13","Clic3","Spdef","Creb3l1","Muc5b","Gp2","Foxj1","Ano1"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,group.by="specific_type",cells.use = E16_Dec7v3_epi@cell.names[!(E16_Dec7v3_epi@meta.data$res.1.4 %in% c(14,17,18))],cex.row = 30,group.cex = 50,group.order = c("Basal-Sostdc1","CyclingBasal","Basal","Basal/Secretory","CyclingSecretory","Secretory-Krt4","Secretory","Ciliated")
  )
```


```{r}
save(E16_Dec7v3_epi,file="E16_Dec7v3_epi.RData")
```
#### different populations of basal cells:
```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
E16_Dec_epi_res14_0over1<-FindMarkers(E16_Dec7v3_epi,ident.1=c(0),ident.2 = c(1),only.pos = TRUE)
E16_Dec_epi_res14_0over1
```
```{r}
write.table(E16_Dec_epi_res14_0over1,"epiSubset_res14_c0overC1.txt",sep="\t")

```
```{r}
library(ggrepel)
```
```{r}
E16_Dec_epi_res14_0over1$gene<-rownames(E16_Dec_epi_res14_0over1)
E16_Dec_epi_res14_0over1$sig<-E16_Dec_epi_res14_0over1$p_val_adj<0.001
volc = ggplot(E16_Dec_epi_res14_0over1, aes(avg_logFC, -log10(p_val_adj))) + #volcanoplot with avg_logFC versus p_val_adj
    geom_point(aes(col=sig)) + #add points colored by significance
    scale_color_manual(values=c("black", "red")) + 
    ggtitle("E16_epi_0over1") 

E16_Dec_epi_res14_0over1_volc.1 <- volc + geom_text_repel(data=head(E16_Dec_epi_res14_0over1, 20), aes(label=gene), point.padding = 1, box.padding = .3) +
  labs(y = expression(-log[10]*" "*"adjusted pvalue"), x = "avg log fold change") + 
  theme(legend.title = element_blank(), legend.position = "top") + 
  scale_fill_discrete(labels = c("Not Sig", "adjusted pval < 0.001"))
```
```{r}
E16_Dec_epi_res14_0over1_volc.1
```

```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
E16_Dec_epi_res14_1over0<-FindMarkers(E16_Dec7v3_epi,ident.1=c(1),ident.2 = c(0),only.pos = TRUE)
E16_Dec_epi_res14_1over0
```
```{r}
write.table(E16_Dec_epi_res14_1over0,"epiSubset_res14_c1overC0.txt",sep="\t")

```

```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
E16_Dec_epi_res14_9over1<-FindMarkers(E16_Dec7v3_epi,ident.1=c(9),ident.2 = c(1),only.pos = TRUE)
E16_Dec_epi_res14_9over1
```
```{r}
write.table(E16_Dec_epi_res14_9over1,"epiSubset_res14_c9overC1.txt",sep="\t")

```

```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
E16_Dec_epi_res14_1over9<-FindMarkers(E16_Dec7v3_epi,ident.1=c(1),ident.2 = c(9),only.pos = TRUE)
E16_Dec_epi_res14_1over9
```
```{r}
write.table(E16_Dec_epi_res14_1over9,"epiSubset_res14_c1overC9.txt",sep="\t")

```
```{r,fig.height=8,fig.width=30}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Epcam","Krt8","Trp63","Krt5","Mki67","Top2a","Smoc2","Ccl20","Sostdc1","Bmp7","Clic3","Cldn10","Tspan33","Ehf","Sfta2","Crip2","Msln","Cyp2s1","Cldn3","Cldn7","Cldn4","AU021092","Tspan1","Chad","Tspan13","Klk10","Klk11","Klk13","Ces1d","Krt4","Krt13","Creb3l1","Spdef"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,cells.use = E16_Dec7v3_epi@cell.names[E16_Dec7v3_epi@meta.data$res.1.4 %in% c(0,13,1,3,9)],group.order = c(0,13,1,3,9),cex.row = 20,group.cex = 30
  )
```
##### different populations of ciliated cells:
```{r,fig.height=5,fig.width=16}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")

DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Foxj1","Shisa8","Mcidas","Ccno","Plk4","Hyls1","Foxn4","Lrrc23","Prr18","Cfap53","Cdhr3","Cdhr4","Ldlrad1","Sntn","Ifitm1","Lbp","Ly6c1","Ly6a"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,cells.use = E16_Dec7v3_epi@cell.names[E16_Dec7v3_epi@meta.data$res.1.4 %in% c(7,11,15)],group.order = c(7,11,15),group.cex = 30,cex.row = 20
  )
```
##### different populations of secretory cells:
```{r,fig.height=10,fig.width=30}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
DoHeatmap(object = E16_Dec7v3_epi, genes.use = c("Epcam","Mki67","Top2a","Sftpb","Hp","Krt4","Krt13","Lgals3","Clic3","Sec14l3","Krt8","Krt18","Muc16","Creb3l1","Spdef","Muc5ac","Muc5b","Foxa3","Fkbp11","Galnt12","Ltf","Gp2","Tff2","Galnt6","Ptgdr","Cd177"),
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T,cells.use = E16_Dec7v3_epi@cell.names[E16_Dec7v3_epi@meta.data$res.1.4 %in% c(6,2,4,5,12,8,10,16)],group.order = c(12,6,4,2,10,5,8,16),cex.col=1,cex.row = 20,group.cex = 30
  )
```



```{r}
epi_marker_c0_wt_mut<-FindMarkers(E16_Dec7v3_epi,cells.1<-WhichCells(object=E16_Dec7v3_epi,ident=0,cells.use = E16_Dec7v3_epi@meta.data$genotype=="wt"),cells.2<-WhichCells(object=E16_Dec7v3_epi,ident=0,cells.use = E16_Dec7v3_epi@meta.data$genotype=="mut"),only.pos = TRUE)
epi_marker_c0_wt_mut
```
##### Sostdc1 seems to be downregulated in mutant basal-Sostdc1 population.

```{r}
epi_marker_c0_mut_wt<-FindMarkers(E16_Dec7v3_epi,cells.1<-WhichCells(object=E16_Dec7v3_epi,ident=0,cells.use = E16_Dec7v3_epi@meta.data$genotype=="mut"),cells.2<-WhichCells(object=E16_Dec7v3_epi,ident=0,cells.use = E16_Dec7v3_epi@meta.data$genotype=="wt"),only.pos = TRUE)
epi_marker_c0_mut_wt
```


```{r}
E16_Dec7v3_epi@meta.data$type_genotype<-as.factor(paste(E16_Dec7v3_epi@meta.data$cell_type,E16_Dec7v3_epi@meta.data$genotype,sep="_"))
```
```{r,fig.height=6,fig.width=6}
E16_Dec7v3_epi<-SetAllIdent(object = E16_Dec7v3_epi, id = "type_genotype")

E16_Dec7v3_epi@ident=factor(E16_Dec7v3_epi@ident,levels(E16_Dec7v3_epi@ident)[c(1,4,2,3,9,10,5,6,7,8)])  #just to reorder the groups
DotPlot(object = E16_Dec7v3_epi, cols.use = c("yellow","red"),genes.plot = rev(c("Sftpa1","Muc5ac","Muc2","Muc20","Muc5b","Muc1","Muc16","Muc4","Pigr","Ltf","Lyz2","Slpi","Lcn2","Sftpd","Sftpb","Defb1")),x.lab.rot = T,plot.legend = T,group.by = "ident",do.return=T)+rotate()+ theme(axis.text.x = element_text(angle = 45, vjust = 1,hjust=1)) 
```
```{r,fig.height=6,fig.width=6}
E16_Dec7v3_epi<-SetAllIdent(object = E16_Dec7v3_epi, id = "type_genotype")

E16_Dec7v3_epi@ident=factor(E16_Dec7v3_epi@ident,levels(E16_Dec7v3_epi@ident)[c(1,4,2,3,9,10,5,6,7,8)])
DotPlot(object = E16_Dec7v3_epi, cols.use = c("forestgreen","magenta3"),genes.plot = rev(c("Nfkbia","Nfkbiz","Retnla","Cxcl17","Cxcl15","Ccl20","Areg","Muc5b","Muc4","Pigr","Ltf","Lyz2","Slpi","Lcn2","Sftpd","Sftpb","Defb1","Lgals3","Itln1")),x.lab.rot = T,plot.legend = T,group.by = "ident",do.return=T,col.min = -2,col.max = 2)+rotate()+ theme(axis.text.x = element_text(angle = 45, vjust = 1,hjust=1)) 
```
```{r}
DE_E16_ciliated_genotype<-FindMarkers(E16_Dec7v3_epi,cells.1<-WhichCells(object=E16_Dec7v3_epi,cells.use = (E16_Dec7v3_epi@meta.data$genotype=="wt" & E16_Dec7v3_epi@meta.data$cell_type=="Ciliated" )),cells.2<-WhichCells(object=E16_Dec7v3_epi,cells.use = (E16_Dec7v3_epi@meta.data$genotype=="mut" & E16_Dec7v3_epi@meta.data$cell_type=="Ciliated" )),only.pos = F,logfc.threshold=0,min.pct=0.05)
DE_E16_ciliated_genotype
```

```{r}
write.table(DE_E16_ciliated_genotype,"DE_E16_ciliated_genotype.txt",sep="\t")
```


```{r}
DE_E16_ciliated_genotype$gene<-rownames(DE_E16_ciliated_genotype)
E16_ciliated_automatic_geneList<-DE_E16_ciliated_genotype$gene[DE_E16_ciliated_genotype$p_val_adj<0.001 & abs(DE_E16_ciliated_genotype$avg_logFC)>0.5 & abs(DE_E16_ciliated_genotype$pct.1-DE_E16_ciliated_genotype$pct.2)>0.15]
```
```{r}
library(ggrepel)
```
```{r,fig.height=8,fig.width=12}

#DE_P4_secretory_genotype$sig<-DE_P4_secretory_genotype$p_val_adj<0.001
DE_E16_ciliated_genotype$threshold<- ifelse(DE_E16_ciliated_genotype$avg_logFC>0 & DE_E16_ciliated_genotype$p_val_adj<0.001, "wt_enrich",ifelse(DE_E16_ciliated_genotype$avg_logFC<0 & DE_E16_ciliated_genotype$p_val_adj<0.001, "mut_enrich","NotSignificant" ) )
ggplot(DE_E16_ciliated_genotype, aes(avg_logFC, -log10(p_val_adj))) + #volcanoplot with avg_logFC versus p_val_adj
    geom_point(aes(col=threshold),size=0.2) + #add points colored by significance
  scale_color_manual(values=c("green", "black","magenta"))+
    ggtitle("E16Ciliated_wt/mut") + geom_text_repel(data=DE_E16_ciliated_genotype[DE_E16_ciliated_genotype$gene %in% E16_ciliated_automatic_geneList,], aes(label=gene), point.padding = 1, box.padding = .3) +
  labs(y = expression(-log[10]*" "*"adjusted pvalue"), x = "avg log fold change") + 
  theme(legend.title = element_blank(), legend.position = "top") 
```
##### PCD genes are not significantly differentially expressed between genotypes:
```{r,fig.height=8,fig.width=12}

#DE_P4_secretory_genotype$sig<-DE_P4_secretory_genotype$p_val_adj<0.001
DE_E16_ciliated_genotype$threshold<- ifelse(DE_E16_ciliated_genotype$avg_logFC>0 & DE_E16_ciliated_genotype$p_val_adj<0.001, "wt_enrich",ifelse(DE_E16_ciliated_genotype$avg_logFC<0 & DE_E16_ciliated_genotype$p_val_adj<0.001, "mut_enrich","NotSignificant" ) )
ggplot(DE_E16_ciliated_genotype, aes(avg_logFC, -log10(p_val_adj))) + #volcanoplot with avg_logFC versus p_val_adj
    geom_point(aes(col=threshold),size=0.2) + #add points colored by significance
  scale_color_manual(values=c("green", "black","magenta"))+
    ggtitle("E16Ciliated_wt/mut") + geom_text_repel(data=DE_E16_ciliated_genotype[DE_E16_ciliated_genotype$gene %in% geneList$Primary.ciliary.dyskinesia,], aes(label=gene), point.padding = 1, box.padding = .3) +
  labs(y = expression(-log[10]*" "*"adjusted pvalue"), x = "avg log fold change") + 
  theme(legend.title = element_blank(), legend.position = "top") 
```

```{r}

DE_E16_ciliated_genotype[DE_E16_ciliated_genotype$gene %in% geneList$Primary.ciliary.dyskinesia,]
```
```{r}

DE_E16_ciliated_genotype[DE_E16_ciliated_genotype$gene %in% geneList$Ciliopathy,]
```


```{r}
df_E16_epi<-FetchData(E16_Dec7v3_epi,c("Spdef","Creb3l1","Scgb3a2","Scgb1a1","Krt4","Krt13","Foxa3","Aqp3","Aqp4","Aqp5","Gp2","Sostdc1","Smoc2","Krt14","Krt15","Krt5","Rac2","Clic3","res.1.2","res.1.4","genotype","seq_group","specific_type","cell_type","Defb1","Lyz2","Ltf","Sftpa1","Sftpd","Sftpb","Slpi","Lcn2","Pigr","Muc5b","Muc5ac","Muc1","Muc2","Muc4","Muc16","Muc20","Lbp","Cd14","Tlr4","Tlr2","Myd88","Ticam1","Itln1","Lgals3","Reg3g","Nod1","Nod2","Ddx58","Ifih1","Dhx58","Ccl5","Cxcl10","Cxcl2","Cxcl1","Pf4","Cxcl12","Cxcl14","Cxcl15","Cxcl16","Cxcl17","Ccl2","Ccl7","Ccl17","Ccl20","Ccl21a","Ccl25","Ccl27a","Ccl28","Cx3cl1","Il10","Tnf","S100a8","S100a9","Il6","Il18","Il1b","Il1rl1","Ccl11","Ccl24","Il33","Il25","Tslp","F2rl1","Retnla","Alox15","Alox5","Gata2","Tgfb2","Tgfb1","Ormdl3","Ptges","Ptgds","Ptgs2","Hpgds","Tbxas1","Areg","Ifnk","Ifnlr1","Nfkbiz","Nfkbia"))

```
##### MicrobialSensing:
```{r, fig.height=3, fig.width=7}
for (i in c("Lbp","Cd14","Tlr4","Tlr2","Myd88","Ticam1","Itln1","Reg3g","Lgals3","Nod1","Nod2","Ddx58","Ifih1","Dhx58"))
{
pdf(file = paste("Manuscript/MicrobialSensing_genotype/E16/",i,".pdf", sep = ""), width = 6, height = 5)
print(ggplot(df_E16_epi,aes_string(x="genotype",y=i))+facet_grid(.~cell_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0)))
dev.off()
}
```

##### antimicrobial effectors:
```{r, fig.height=3, fig.width=7}
for (i in c("Muc1","Muc4","Muc16","Muc20","Muc5b","Muc5ac","Muc2","Defb1","Lyz2","Ltf","Sftpa1","Sftpd","Sftpb","Slpi","Lcn2","Pigr","Chil4"))
{
pdf(file = paste("Manuscript/Effectors_genotype/E16/",i,".pdf", sep = ""), width = 6, height = 5)
print(ggplot(df_E16_epi,aes_string(x="genotype",y=i))+facet_grid(.~cell_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0)))
dev.off()
}
```
##### chemokines:
```{r, fig.height=3, fig.width=7}
for (i in c("Ccl5","Cxcl10","Cxcl2","Cxcl1","Pf4","Cxcl12","Cxcl14","Cxcl15","Cxcl16","Cxcl17","Ccl2","Ccl7","Ccl17","Ccl20","Ccl21a","Ccl25","Ccl27a","Ccl28","Cx3cl1"))
{
pdf(file = paste("Manuscript/chemokines_genotype/E16/",i,".pdf", sep = ""), width = 6, height = 5)
print(ggplot(df_E16_epi,aes_string(x="genotype",y=i))+facet_grid(.~cell_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0)))
dev.off()
}
```
##### Th2:
```{r, fig.height=3, fig.width=7}
for (i in c("Il10","Tnf","S100a8","S100a9","Il6","Il18","Il1b","Il1rl1","Ccl11","Ccl24","Il33","Il25","Tslp","F2rl1","Retnla","Alox15","Alox5","Gata2","Tgfb2","Tgfb1","Ormdl3","Ptges","Ptgds","Ptgs2","Hpgds","Tbxas1","Areg"))
{
pdf(file = paste("Manuscript/Th2_genotype/E16/",i,".pdf", sep = ""), width = 6, height = 5)
print(ggplot(df_E16_epi,aes_string(x="genotype",y=i))+facet_grid(.~cell_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0)))
dev.off()
}
```


```{r, fig.height=5, fig.width=10}
library(ggpubr)

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Muc2))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```


```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Sftpb))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```
```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Scgb3a2))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1))
```
```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Scgb1a1))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```
```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Sostdc1))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```
```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$specific_type=="Doublet"),],aes(genotype,Smoc2))+facet_grid(.~specific_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```

```{r, fig.height=5, fig.width=10}

ggplot(df_E16_epi[!(df_E16_epi$cell_type=="Doublet"),],aes(genotype,Nfkbiz))+facet_grid(.~cell_type)+geom_dotplot(binaxis="y",aes(fill=genotype),binwidth=0.05,stackdir="center",position=position_dodge(0.8), dotsize=0.2)+stat_compare_means(comparisons = list(c("wt", "mut")),method="wilcox.test",size=4,label="p.adj")+ stat_summary(aes(color=genotype),fun.data=mean_sdl, fun.args = list(mult=1), 
                 geom="pointrange",position=position_dodge(0.7))+ theme(axis.text.x = element_text(angle = 45,hjust=1),strip.text.x = element_text(size = 9, colour = "black", angle = 0))
```


##### explore genes correlated with basal-->secretory changes:
##### c3 is the basal/secretory population
```{r}
E16_Dec7v3_epi <- SetAllIdent(object = E16_Dec7v3_epi, id = "res.1.4")
E16_Dec7v3_epi_subc3<-SubsetData(object=E16_Dec7v3_epi,ident.use=c(3))
table(E16_Dec7v3_epi_subc3@meta.data$res.1.4)
```
```{r}
E16_Dec7v3_epi_subc3@meta.data<-E16_Dec7v3_epi_subc3@meta.data[,-which(names(E16_Dec7v3_epi_subc3@meta.data) %in% c("res.0.8", "res.1.2", "res.1.4", "res.1.6"))] #remove old metadata
E16_Dec7v3_epi_subc3 <- ScaleData(object = E16_Dec7v3_epi_subc3)
```
```{r}
E16_Dec7v3_epi_subc3 <- FindVariableGenes(object = E16_Dec7v3_epi_subc3, do.plot = TRUE, x.low.cutoff=0.1,x.high.cutoff = Inf, y.cutoff = 0.5)
```

```{r}
E16_Dec7v3_epi_subc3 <- RunPCA(object = E16_Dec7v3_epi_subc3, do.print = FALSE)
PCAPlot(E16_Dec7v3_epi_subc3)
```

```{r}
E16_Dec7v3_epi_subc3 <- ProjectPCA(object = E16_Dec7v3_epi_subc3, do.print = TRUE)
```

```{r}
PCElbowPlot(object = E16_Dec7v3_epi_subc3)
```
```{r}
n.pcs.sub3 = 13
```
#####resolution parameter sets the ‘granularity’ of the downstream clustering, with increased values leading to a greater number of clusters. 
```{r}
res.used <- 1.2
```

```{r}
E16_Dec7v3_epi_subc3 <- FindClusters(object = E16_Dec7v3_epi_subc3, reduction.type = "pca", dims.use = 1:n.pcs.sub3, 
                     resolution = res.used, print.output = 0, force.recalc = T)
```
```{r}
E16_Dec7v3_epi_subc3 <- RunTSNE(object = E16_Dec7v3_epi_subc3, dims.use = 1:n.pcs.sub3, perplexity=30)
```
```{r}
TSNEPlot(object = E16_Dec7v3_epi_subc3, do.label = T)
```
```{r}
res.used <- 0.8
```
```{r}
TSNEPlot(object = E16_Dec7v3_epi_subc3, do.label = T)
```

```{r}
res.used <- 0.6
```
```{r}
TSNEPlot(object = E16_Dec7v3_epi_subc3, do.label = T)
```
```{r,fig.height=8,fig.width=30}
E16_Dec7v3_epi_subc3 <- SetAllIdent(object = E16_Dec7v3_epi_subc3, id = "res.0.6")
DoHeatmap(object = E16_Dec7v3_epi_subc3, genes.use = c("Epcam","Krt8","Trp63","Krt5","Mki67","Top2a","Smoc2","Ccl20","Sostdc1","Bmp7","Clic3","Cldn10","Tspan33","Ehf","Sfta2","Crip2","Msln","Cyp2s1","Cldn3","Cldn7","Cldn4","AU021092","Tspan1","Chad","Tspan13","Klk10","Klk11","Klk13","Ces1d","Krt4","Krt13","Creb3l1"), 
    slim.col.label = TRUE, group.label.rot = TRUE,use.scaled = T
  )
```
```{r}
cor(E16_Dec7v3_epi_subc3@scale.data["Trp63",], E16_Dec7v3_epi_subc3@scale.data["Creb3l1",])
```
```{r}
cor(E16_Dec7v3_epi_subc3@data["Trp63",], E16_Dec7v3_epi_subc3@data["Spdef",])
```
```{r}
Trp63_cor<-apply(E16_Dec7v3_epi_subc3@scale.data, 1, function(x) cor(E16_Dec7v3_epi_subc3@scale.data["Trp63",],x)) 
```
```{r}
head(Trp63_cor) 
```
```{r}
min(Trp63_cor,na.rm = T) 
```
```{r}
Trp63_cor_order<-order(Trp63_cor,decreasing=T) 
```
```{r}
head(Trp63_cor[Trp63_cor_order],20)
```
```{r}
sum(is.na(Trp63_cor[Trp63_cor_order]))
```
```{r}
Trp63_cor[Trp63_cor_order][15706:15686]
```
```{r}
tail(Trp63_cor[Trp63_cor_order],20)
```



```{r}
cor.test(E16_Dec7v3_epi_subc3@scale.data["Spdef",], E16_Dec7v3_epi_subc3@scale.data["Trp63",],method="pearson")
```

```{r}
cor.test(E16_Dec7v3_epi_subc3@scale.data["Spdef",], E16_Dec7v3_epi_subc3@scale.data["Trp63",],method="kendall")
```

```{r}
Trp63_cor_test<-apply(E16_Dec7v3_epi_subc3@scale.data, 1, function(x) cor.test(E16_Dec7v3_epi_subc3@scale.data["Trp63",],x,method="pearson")) 
```
```{r}
df_Trp63_corTest<-as.data.frame(do.call(rbind, Trp63_cor_test))

```
```{r}
df_order_Trp63_corTest<-df_Trp63_corTest[order(unlist(df_Trp63_corTest$estimate)),]
```
```{r}
tidy_Trp63_cor<-cbind(df_order_Trp63_corTest$estimate,df_order_Trp63_corTest$p.value)
```

```{r}
colnames(tidy_Trp63_cor)<-c("cor","p.value")
```

```{r}
head(tidy_Trp63_cor,20)
```
```{r}
tidy_Trp63_cor[15706:15686,]
```
```{r}
tail(tidy_Trp63_cor)
```

```{r}
Krt8_cor_test<-apply(E16_Dec7v3_epi_subc3@scale.data, 1, function(x) cor.test(E16_Dec7v3_epi_subc3@scale.data["Krt8",],x,method="pearson")) 
```

```{r}
df_Krt8_corTest<-as.data.frame(do.call(rbind, Krt8_cor_test))

```
```{r}
df_order_Krt8_corTest<-df_Krt8_corTest[order(unlist(df_Krt8_corTest$estimate)),]
```
```{r}
sum(is.na(df_order_Krt8_corTest$estimate))
```

```{r}
tidy_Krt8_cor<-cbind(df_order_Krt8_corTest$estimate,df_order_Krt8_corTest$p.value)
```

```{r}
colnames(tidy_Krt8_cor)<-c("cor","p.value")
```

```{r}
head(tidy_Krt8_cor,20)
```
```{r}
tidy_Krt8_cor[15666:15706,]
```

```{r}
cor.test(E16_Dec7v3_epi_subc3@scale.data["Krt8",E16_Dec7v3_epi_subc3@meta.data$genotype=="wt"], E16_Dec7v3_epi_subc3@scale.data["Numb",E16_Dec7v3_epi_subc3@meta.data$genotype=="wt"],method="pearson")
```
```{r}
cor.test(E16_Dec7v3_epi_subc3@scale.data["Krt8",E16_Dec7v3_epi_subc3@meta.data$genotype=="mut"], E16_Dec7v3_epi_subc3@scale.data["Numb",E16_Dec7v3_epi_subc3@meta.data$genotype=="mut"],method="pearson")
```
















